File: p1-121/p1e1/src/EcuacionLineal.java
import java.util.Scanner;

public class EcuacionLineal {
    private float a;
    private float b;
    private float c;
    private float d;
    private float e;
    private float f;
    private double x;
    private double y;

    public EcuacionLineal(double a, double b, double c, double d, double e, double f) {
        this.a = (float) a;
        this.b = (float) b;
        this.c = (float) c;
        this.d = (float) d;
        this.e = (float) e;
        this.f = (float) f;
    }

    public void leer(){
        Scanner lee = new Scanner(System.in);
        System.out.println("Ingrese a, b, c, d, e, f: ");
        a = lee.nextFloat();
        b = lee.nextFloat();
        c = lee.nextFloat();
        d = lee.nextFloat();
        e = lee.nextFloat();
        f = lee.nextFloat();
        lee.close();
    }

    public void X(){
        x = ((e*d)-(b*f))/((a*d)-(b*c));
    }
    public void Y(){
        y = ((a*f)-(e*c))/((a*d)-(b*c));
    }


    public void tieneSolucion() {
        if ((a*d)-(b*c) != 0){
            X();
            Y();
            System.out.println(x + ", " + y);
        }
        else
            System.out.println("La ecuación no tiene solucion");
    }
}

File: p1-121/py1e1/EcuacionLineal.py
class EcuacionLineal:
    def __init__(self, a, b, c, d, e, f):
        self.a = a
        self.b = b
        self.c = c
        self.d = d
        self.e = e
        self.f = f
        self.x = None
        self.y = None

    def X(self):
        self.x = ((self.e*self.d)-(self.b*self.f))/((self.a*self.d)-(self.b*self.c))

    def Y(self):
        self.y = ((self.a*self.f)-(self.e*self.c))/((self.a*self.d)-(self.b*self.c))

    def tieneSolucion(self):
        if((self.a*self.d)-(self.b*self.c) != 0):
            self.X()
            self.Y()
            print(self.x, ", ",self.y)
        else:
            print("la escuacion no tiene solucion")
try:
    v = input("Ingrese a, b, c, d, e, f: ").split()
    a, b, c, d, e, f = map(float, v)

    ec1 = EcuacionLineal(a, b, c, d, e, f)
    ec1.tieneSolucion()
except ValueError:
    print("Error")